lib/ansible/modules/network/netscaler/netscaler_gslb_vserver.py

```diff
@@ -0,0 +1,946 @@
+
+    netmask:
+        description:
+            - "IPv4 network mask for use in the SOURCEIPHASH load balancing method."
+            - "Minimum length = 1"
+
+    v6netmasklen:
+        description:
+            - >-
+                Number of bits to consider, in an IPv6 source IP address, for creating the hash that is required by
+                the C(SOURCEIPHASH) load balancing method.
+            - "Default value: C(128)"
+            - "Minimum value = C(1)"
+            - "Maximum value = C(128)"
+
+    tolerance:
+        description:
+            - >-
+                Site selection tolerance, in milliseconds, for implementing the RTT load balancing method. If a
+                site's RTT deviates from the lowest RTT by more than the specified tolerance, the site is not
+                considered when the NetScaler appliance makes a GSLB decision. The appliance implements the round
+                robin method of global server load balancing between sites whose RTT values are within the specified
+                tolerance. If the tolerance is 0 (zero), the appliance always sends clients the IP address of the
+                site with the lowest RTT.
+            - "Minimum value = C(0)"
+            - "Maximum value = C(100)"
+
+    persistencetype:
+        choices:
+            - 'SOURCEIP'
+            - 'NONE'
+        description:
+            - "Use source IP address based persistence for the virtual server."
+            - >-
+                After the load balancing method selects a service for the first packet, the IP address received in
+                response to the DNS query is used for subsequent requests from the same client.
+
+    persistenceid:
+        description:
+            - >-
+                The persistence ID for the GSLB virtual server. The ID is a positive integer that enables GSLB sites
+                to identify the GSLB virtual server, and is required if source IP address based or spill over based
+                persistence is enabled on the virtual server.
+            - "Minimum value = C(0)"
+            - "Maximum value = C(65535)"
+
+    persistmask:
+        description:
+            - >-
+                The optional IPv4 network mask applied to IPv4 addresses to establish source IP address based
+                persistence.
+            - "Minimum length = 1"
+
+    v6persistmasklen:
+        description:
+            - >-
+                Number of bits to consider in an IPv6 source IP address when creating source IP address based
+                persistence sessions.
+            - "Default value: C(128)"
+            - "Minimum value = C(1)"
+            - "Maximum value = C(128)"
+
+    timeout:
+        description:
+            - "Idle time, in minutes, after which a persistence entry is cleared."
+            - "Default value: C(2)"
+            - "Minimum value = C(2)"
+            - "Maximum value = C(1440)"
+
+    mir:
+        choices:
+            - 'ENABLED'
```

Comment:
Yes, since all other modules use uppercase for the 'ENABLED', 'DISABLED' values I think persistency is significant.

If the lower case convention is an important issue, after the last modules are merged, we could make a PR for all submitted modules to fix this for all modules at once.